From 25624879911ea2443ac9b1c4810066e6bfe69c0d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=BCnther=20Noack?= <gnoack@google.com>
Date: Thu, 3 Jul 2025 15:05:35 +0200
Subject: [PATCH 10/33] %%PDFTOPDFCollate comment parsing: Increment p where
 needed

Without this, if strchr succeeds, p will point to a ':' character.
*p will therefore never be a whitespace in the subsequent loop,
or compare successfully against the "true" string.
---
 cupsfilters/ghostscript.c   | 1 +
 cupsfilters/mupdftopwg.c    | 1 +
 cupsfilters/pdftoraster.cxx | 1 +
 3 files changed, 3 insertions(+)

diff --git a/cupsfilters/ghostscript.c b/cupsfilters/ghostscript.c
index 30d0e394..88d19b3a 100644
--- a/cupsfilters/ghostscript.c
+++ b/cupsfilters/ghostscript.c
@@ -109,6 +109,7 @@ parse_pdf_header_options(FILE *fp,
         h->Collate = CUPS_FALSE;
         continue;
       }
+      p ++;
       while (*p == ' ' || *p == '\t')
 	p ++;
       if (strncasecmp(p, "true", 4) == 0)
diff --git a/cupsfilters/mupdftopwg.c b/cupsfilters/mupdftopwg.c
index dcde56ed..9c8484fe 100644
--- a/cupsfilters/mupdftopwg.c
+++ b/cupsfilters/mupdftopwg.c
@@ -102,6 +102,7 @@ parse_pdf_header_options(FILE *fp,
         h->Collate = CUPS_FALSE;
         continue;
       }
+      p ++;
       while (*p == ' ' || *p == '\t')
 	p ++;
       if (strncasecmp(p, "true", 4) == 0)
diff --git a/cupsfilters/pdftoraster.cxx b/cupsfilters/pdftoraster.cxx
index 1d14cd59..d1677757 100644
--- a/cupsfilters/pdftoraster.cxx
+++ b/cupsfilters/pdftoraster.cxx
@@ -446,6 +446,7 @@ parse_pdftopdf_comment(FILE *fp,
 	*deviceCollate = false;
 	continue;
       }
+      p ++;
       while (*p == ' ' || *p == '\t') p ++;
       if (strncasecmp(p, "true", 4) == 0)
 	*deviceCollate = true;
-- 
2.53.0

